File: easy/python/move_zeroes.py
from ast import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left, right = 0, 1
        while right < len(nums):
            if nums[left] == 0 and nums[right] == 0:
                right += 1
            elif nums[left] == 0:
                temp = nums[left]
                nums[left] = nums[right]
                nums[right] = temp
            else:
                left += 1
                right += 1

        return nums

File: easy/ruby/move_zeroes.rb
# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
def move_zeroes(nums)
  left = 0
  right = 1
  while right < nums.length
    next right += 1 if nums[left].zero? && nums[right].zero?

    if nums[left].zero?
      temp = nums[left]
      nums[left] = nums[right]
      nums[right] = temp
    end

    right += 1
    left += 1
  end

  nums
end
